src/plugins/mce/plugin.c

```diff
@@ -62,7 +82,12 @@ toggle_pattern (NCore *core, const char *pattern, gboolean activate)
         return FALSE;
     }
 
-    ret = n_dbus_async_call_full (core, NULL, NULL, DBUS_BUS_SYSTEM, msg);
+    if (activate) {
+        ret = n_dbus_async_call_full (core, activate_cb, g_strdup (pattern), DBUS_BUS_SYSTEM, msg);
```

Comment:
side note: There are all manner of ways n_dbus_async_call_full() can leak the just dupped pattern, but - as this kind of orphan-user-data-on-hiccups behavior is present elsewhere too - maybe we can assume happy path is followed?

Reply:
Internally calls dbus_pending_call_set_notify() which is defined "Sets a notification function to be called when the reply is received or the pending call times out. " so I assumed this would get always called and thus free the pattern. Perhaps on error the pattern shouldn't be marked as playing, but that's another thing.